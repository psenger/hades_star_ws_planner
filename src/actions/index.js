export const ActionTypes = {
     SET_HYDRO : 'SET_HYDRO',
     SET_RP : 'SET_RP',
     SET_GATE : 'SET_GATE'
}

export const setHydro = ( key, id, x, y, value ) => ({
    type: ActionTypes.SET_HYDRO,
    value: {
        key,
        id,
        hydro : { x, y, value }
    }
});
