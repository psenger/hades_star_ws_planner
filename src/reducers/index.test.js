import configureStore from './index';
import {setHydro} from '../actions/index';

describe('reducers', () => {
    describe('gridReducer',()=>{
        describe('setHydro', () => {
            it('should be equal to default state', ( ) => {
                const initialState = Object.freeze({
                });
                const store = configureStore(initialState);
                store.dispatch(setHydro( 'A1', 'a5c0ee1d-9f40-45cc-b568-1bf598ea5a70', 100, 200, 1000 ));
                expect( store.getState() ).toMatchSnapshot();
            });
        });
    });
});
