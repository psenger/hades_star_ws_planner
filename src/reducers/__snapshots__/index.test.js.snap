// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers gridReducer setHydro should be equal to default state 1`] = `
Object {
  "gridReducer": Object {
    "gate": Object {},
    "grid": Object {
      "A1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [
          "a5c0ee1d-9f40-45cc-b568-1bf598ea5a70",
        ],
        "rpIdx": Array [],
        "x": 0,
        "y": 130,
      },
      "A2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 0,
        "y": 217,
      },
      "A3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 0,
        "y": 304,
      },
      "A4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 0,
        "y": 391,
      },
      "B1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 75,
        "y": 87,
      },
      "B2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 75,
        "y": 174,
      },
      "B3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 75,
        "y": 261,
      },
      "B4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 75,
        "y": 348,
      },
      "B5": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 75,
        "y": 435,
      },
      "C1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 43,
      },
      "C2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 130,
      },
      "C3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 217,
      },
      "C4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 304,
      },
      "C5": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 391,
      },
      "C6": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 150,
        "y": 478,
      },
      "D1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 0,
      },
      "D2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 87,
      },
      "D3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 174,
      },
      "D4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 261,
      },
      "D5": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 348,
      },
      "D6": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 435,
      },
      "D7": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 225,
        "y": 522,
      },
      "E1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 43,
      },
      "E2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 130,
      },
      "E3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 217,
      },
      "E4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 304,
      },
      "E5": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 391,
      },
      "E6": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 300,
        "y": 478,
      },
      "F1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 375,
        "y": 87,
      },
      "F2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 375,
        "y": 174,
      },
      "F3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 375,
        "y": 261,
      },
      "F4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 375,
        "y": 348,
      },
      "F5": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 375,
        "y": 435,
      },
      "G1": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 450,
        "y": 130,
      },
      "G2": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 450,
        "y": 217,
      },
      "G3": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 450,
        "y": 304,
      },
      "G4": Object {
        "gateIdx": Array [],
        "hydroIdx": Array [],
        "rpIdx": Array [],
        "x": 450,
        "y": 391,
      },
    },
    "gridIdx": Array [
      Array [
        "A1",
        "A2",
        "A3",
        "A4",
      ],
      Array [
        "B1",
        "B2",
        "B3",
        "B4",
        "B5",
      ],
      Array [
        "C1",
        "C2",
        "C3",
        "C4",
        "C5",
        "C6",
      ],
      Array [
        "D1",
        "D2",
        "D3",
        "D4",
        "D5",
        "D6",
        "D7",
      ],
      Array [
        "E1",
        "E2",
        "E3",
        "E4",
        "E5",
        "E6",
      ],
      Array [
        "F1",
        "F2",
        "F3",
        "F4",
        "F5",
      ],
      Array [
        "G1",
        "G2",
        "G3",
        "G4",
      ],
    ],
    "hydro": Object {
      "a5c0ee1d-9f40-45cc-b568-1bf598ea5a70": Object {
        "value": 1000,
        "x": 100,
        "y": 200,
      },
    },
    "rp": Object {},
  },
}
`;
